File: src/main/java/nuclearcoder/discordbot/TimerKeepAlive.java
package nuclearcoder.discordbot;

import nuclearcoder.discordbot.database.Database;

import java.util.Timer;
import java.util.TimerTask;
import java.util.concurrent.TimeUnit;
import java.util.concurrent.atomic.AtomicBoolean;

public class TimerKeepAlive extends TimerTask {

    private static final long KEEP_ALIVE = TimeUnit.SECONDS.toMillis(10);

    final AtomicBoolean alive = new AtomicBoolean(true);

    private Timer timer;

    TimerKeepAlive()
    {
        this.timer = new Timer("keepalive", false);
        timer.schedule(this, KEEP_ALIVE, KEEP_ALIVE);
    }

    @Override public void run()
    {
        if (alive.get())
        {
            Database.keepAlive();
        }
        else
        {
            timer.cancel();
        }
    }

}

File: src/main/java/nukebot/TimerKeepAlive.kt
package nukebot

import nukebot.database.Database
import java.util.*
import java.util.concurrent.atomic.AtomicBoolean

class TimerKeepAlive : TimerTask() {

    internal val alive = AtomicBoolean(true)

    private val timer = Timer("keep-alive", false)

    init {
        timer.schedule(this, KEEP_ALIVE, KEEP_ALIVE)
    }

    override fun run() {
        if (alive.get()) {
            Database.keepAlive()
        } else {
            timer.cancel()
        }
    }

}
